### DISCUSSION TITLE
为 Daily Note(或者 Daily Document) 设置自动模板

### DISCUSSION THREAD

**Discussion by lovedebug** [5eagull/RemNote-Chinese-chat-group#9] [实用技巧]
每日日记是大家经常打开的页面, 随手记录一些自己的摘录或者感受.
如果我们放任自由, 不按需求组织页面, 其实也变相的变成了一个垃圾桶.
所以, 为每日日记设置模板是一个很好的注意.
关于模板, 大家可以参考 RemNote 的官方文档, 本文也主要是针对 Pro 用户的技巧
有两种设置模板的方式
1. CTRL + P 搜索 'Automatically Add Template'
    - 新增一个 Daily Routing 子项, 使用 / 命令, 然后选择 Apply 'Automatically Template' Power-up
    - 在其内部增加子节点, 并分别使用 / 命令, 选择并应用 Turn into a slot
    - 模板设置好之后就可以在 Daily Notes 上通过设置 ##Daily Routing 标签引入模板
2. CTRL + P 搜索 'Daily Notes'
    - 在 Daily Notes 标题上使用 ##Automatically Template 标签
    - 在 Daily Notes 内部新增所需要的子节点作为, 分别使用 / 命令, 选择并应用 Turn into a slot
    - 每次 ALT + D 新增 Daily note 都会自动应用模板

目前笔者使用的是方案二, 毕竟自动化节省时间, 哈哈

**Comment by makerjackie** [comment]
采用第2种方法，我只能搜索到 daily Document ，搜索不到 daily Note @lovedebug

**Reply by Deng-Jiajun** [reply]
`Daily Document` 就是 `Daily Note`

**Reply by makerjackie** [reply]
thx 懂了

**Comment by zhaolaoer** [comment]
我把模板设置在Automatically Template里面，以用会出现Automatically Template什么什么这个是不是和顶级rem有关

**Comment by zhaolaoer** [comment]
我把模板设置在Automatically Template里面，以用会出现Automatically Template什么什么这个是不是和顶级rem有关

**Reply by lovedebug** [reply]
没明白你的意思. Automatically template 中可以存放 template, 另外这个rem 本身的特点是任何rem如果将其作为标签, 则该 rem 可以自动应用模板, 模板定义在自身 rem 内

**Reply by zhaolaoer** [reply]
不好意思我没说清楚，我在Automatically template设置了一个关于数学性质的模板，在其他rem标记上Automatically template成功出线模板，出现模板同时还要还带有标题性质
![Uploading QQ截图20210515180531.png…]()

**Reply by lovedebug** [reply]
你在 automatically template 设置了模板假设如下
* 定理 => 此处要command 命令选择 apply automatically template
    * 推论 => 此处要 command 命令选择 apply slot
    * 公式 => 此处要 command 命令选择 apply slot
那么在任何 rem 或者 rem 的子项中需要使用的地方直接执行 ##定理 就会应用了

**Reply by zhaolaoer** [reply]
嗯嗯好的

**Reply by zhaolaoer** [reply]
谢谢

**Comment by Deng-Jiajun** [comment]
关于自动模板的问题，总结一下：

当一个标签 `Example Tag` 下含有模板（即类型为 `slot` 的 Rem），且被打上了 `Automatically Add Template` 标签。在使用 `Example Tag` 标签对其他 Rem 进行标记时，RemNote 会自动为这个被标记 `Example Tag` 的 Rem 添加 `Example Tag` 下的所有 Slot Rem（未添加 `Automatically Add Template` 标签时，还需要自己点选以引入 Slot Rem）。

所以实际上 `Automatically Add Template` 就干了一件事：自动添加所有 Slot Rem。大概就是省了几下点击鼠标的工作而已，`Daily Document` 的标签的系统自动添加的，所以节省了「打标签+引入模板」的工作。对于其他文档，只是减少一个全部引入的工作，标签还是要自己打。（要注意一点：如果不是需要所有的子项，这种做法一定程度上还会减少使用的灵活性，因为你还要删掉用不到的模板项）

因为我们在选择创建 `Today's Document`（或者直接按 `Alt` + `D` 创建)的时候，就会自动添加`Daily Document` 标签，因此只要为 `Daily Document` 添加一个 `Automatically Add Template` 标签， `Daily Document` 下添加的模板（即类型为 `slot` 的 Rem）就会被全部写入这个新创建的今天的 Document。

模板不需要 `Automatically Add Template` 下面，随便写在哪里都行，只要打上 `Automatically Add Template` 标签，用这个模板打标签的时候，就会自动添加所有的 `slot`。
